src/sync.cpp

```diff
@@ -182,9 +183,11 @@ static void push_lock(MutexType* c, const CLockLocation& locklocation)
 
         const LockPair p2 = std::make_pair(c, i.first);
         if (lockdata.lockorders.count(p2)) {
-            auto lock_stack_copy = lock_stack;
+            auto lock_stack_current = lock_stack;
```

Comment:
In commit "Release `LockData::dd_mutex` before calling `*_detected` functions" (08cfe273ddf9bd7640456865926ec9d94b363520)

Previous name `lock_stack_copy` seems like it explains the purpose for this variable better than `lock_stack_current`. Also `lock_stack_copy` is the name used above and it would be nicer if double_lock_detected and potential_deadlock_detected cases were handled consistently I think, unless I am missing something.